controllers/kustomization_impersonation.go

```diff
@@ -177,15 +177,20 @@ func (ki *KustomizeImpersonation) getKubeConfig(ctx context.Context) ([]byte, er
 	}
 
 	var kubeConfig []byte
-	for k := range secret.Data {
-		if k == "value" || k == "value.yaml" {
-			kubeConfig = secret.Data[k]
-			break
+	switch {
+	case ki.kustomization.Spec.KubeConfig.SecretRef.Key != "":
+		key := ki.kustomization.Spec.KubeConfig.SecretRef.Key
+		kubeConfig = secret.Data[key]
+		if kubeConfig == nil {
+			return nil, fmt.Errorf("KubeConfig secret '%s' does not contain a '%s' key with a kubeconfig", secretName, key)
```

Comment:
```suggestion
			return nil, fmt.Errorf("KubeConfig secret '%s' does not contain a '%s' key with data", secretName, key)
```

(we _assume_ it to be a kubeconfig, but we are not sure)

Same for error below.

Reply:
Updated